### ISSUE TITLE
Prevent saving of default values

### ISSUE THREAD

**Issue by pacifier007** [CMB2/CMB2#1068]
Please tag this [help wanted].

I am new to CMB2 so I might be missing something but I have noticed upon saving, all values are saved in the options / meta array regardless whether the value is default or not. While it's understandable for most setups, I wish to have the ability to change the defaults by just changing the 'default' value set for a field in the array - in future updates of a plugin/theme. 

In short, I don't want to save the field value in database at all unless the value is different from 'default'. Can someone point me to any filters related to saving data that I can use to change this behavior? Or if there's an extension/plugin that does the same.

**Comment by jtsternberg** [comment]
To do that, you could probably use the [`escape_cb`](https://github.com/CMB2/CMB2/wiki/Field-Parameters#escape_cb), and the [`sanitization_cb`](https://github.com/CMB2/CMB2/wiki/Field-Parameters#sanitization_cb) parameters. Something like:
```php
function my_sanitization_cb( $value, $field_args, $field ) {
	if ( 'whatever_default_value' === $value ) {
		$value = '';
	}

	// Be sure to sanitize the data, since we're bypassing the default sanitization.
	$value = sanitize_text_field( $value );

	return $value;
}

function my_escape_cb( $value, $field_args, $field ) {
	if ( empty( $value ) ) {
		$value = 'whatever_default_value';
	}

	// Be sure to escape the data, since we're bypassing the default escaping.
	$value = esc_attr( $value );

	return $value;
}
```

**Comment by pacifier007** [comment]
Thanks for pointing these out. I did come across them while reading the documentation,  unfortunately they seem to be usable on one-field basis. I would like to have that behavior globally - for all fields. Upon saving, for all fields, I wish to check if the value is the [default value](https://github.com/CMB2/CMB2/wiki/Field-Parameters#default) and prevent saving it. Is there a global "pre-save" filter somewhere?

**Comment by jtsternberg** [comment]
You can use this filter on save: https://github.com/CMB2/CMB2/blob/trunk/includes/CMB2_Field.php#L495
And this filter on display: https://github.com/CMB2/CMB2/blob/trunk/includes/CMB2_Field.php#L716
